### PR TITLE
Franka allegro

### PR DESCRIPTION
**Pull request by wensi-ai**

### CODE REVIEW

**Review by cremebrule — APPROVED** [review]
lgtm!

File: omnigibson/robots/franka_allegro.py [hunk 1]
```diff
@@ -0,0 +1,251 @@
+import os
+import numpy as np
+
+from omnigibson.macros import gm
+from omnigibson.robots.manipulation_robot import ManipulationRobot
+
+RESET_JOINT_OPTIONS = {
```
Comment (cremebrule) [inline omnigibson/robots/franka_allegro.py:7]: remove

File: omnigibson/robots/franka.py [hunk 2]
```diff
@@ -0,0 +1,229 @@
...
+            kwargs (dict): Additional keyword arguments that are used for other super() calls from subclasses, allowing
+                for flexible compositions of various object subclasses (e.g.: Robot is USDObject + ControllableObject).
+        """
+        # Parse reset joint pos if specifying special string
+        if isinstance(reset_joint_pos, str):
+            assert (
+                reset_joint_pos in RESET_JOINT_OPTIONS
+            ), "reset_joint_pos should be one of {} if using a string!".format(RESET_JOINT_OPTIONS)
+            reset_joint_pos = (
+                self.tucked_default_joint_pos if reset_joint_pos == "tuck" else self.untucked_default_joint_pos
```
Comment (cremebrule) [inline omnigibson/robots/franka.py:106]: update

File: omnigibson/robots/franka_allegro.py [hunk 3]
```diff
@@ -0,0 +1,251 @@
...
+            kwargs (dict): Additional keyword arguments that are used for other super() calls from subclasses, allowing
+                for flexible compositions of various object subclasses (e.g.: Robot is USDObject + ControllableObject).
+        """
+        # Parse reset joint pos if specifying special string
+        if isinstance(reset_joint_pos, str):
+            assert (
+                reset_joint_pos in RESET_JOINT_OPTIONS
+            ), "reset_joint_pos should be one of {} if using a string!".format(RESET_JOINT_OPTIONS)
+            reset_joint_pos = (
+                self.tucked_default_joint_pos if reset_joint_pos == "tuck" else self.untucked_default_joint_pos
```
Comment (cremebrule) [inline omnigibson/robots/franka_allegro.py:105]: update

File: omnigibson/robots/franka_allegro.py (hunk 1 above)
Comment (wensi-ai) [reply omnigibson/robots/franka_allegro.py]: done!

File: omnigibson/robots/franka.py [hunk 4]
```diff
@@ -0,0 +1,229 @@
+import os
+import numpy as np
+
+from omnigibson.macros import gm
+from omnigibson.robots.manipulation_robot import ManipulationRobot
+
+
+RESET_JOINT_OPTIONS = {
```
[the remark this answers is not in the document]
Comment (wensi-ai) [reply omnigibson/robots/franka.py]: done

File: omnigibson/robots/franka.py (hunk 2 above)
Comment (wensi-ai) [reply omnigibson/robots/franka.py]: done

File: omnigibson/robots/franka_allegro.py (hunk 3 above)
Comment (wensi-ai) [reply omnigibson/robots/franka_allegro.py]: done

**Pull request merged**